specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -2127,6 +2416,122 @@
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "the name of the Enterprise Channel for which availability needs to be checked.",
+          "type": "string"
+        }
+      }
+    },
+    "EnterpriseChannelCheckNameAvailabilityResponseBody": {
+      "description": "The response body returned for a request to Bot Service Management to check availability of an Enterprise Channel name.",
+      "type": "object",
+      "properties": {
+        "valid": {
+          "description": "indicates if the Enterprise Channel name is valid.",
+          "type": "boolean"
+        },
+        "message": {
+          "description": "additional message from the bot management api showing why a bot name is not available",
+          "type": "string"
+        }
+      }
+    },
+    "EnterpriseChannelResponseList": {
+      "properties": {
+        "nextLink": {
+          "description": "The link used to get the next page of bot service resources.",
+          "type": "string"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnterpriseChannel"
+          },
+          "description": "The list of Enterprise Channels and their properties."
+        }
+      },
+      "description": "The list of  bot service operation response."
+    },
+    "EnterpriseChannel": {
+      "type": "object",
+      "description": "Enterprise Channel resource definition",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EnterpriseChannelProperties",
+          "description": "The set of properties specific to an Enterprise Channel resource"
+        }
+      }
+    },
+    "EnterpriseChannelProperties": {
+      "properties": {
+        "state": {
+          "type": "string",
+          "description": "The current state of the Enterprise Channel",
+          "enum": ["Creating", "CreateFailed", "Started", "Starting", "StartFailed", "Stopped", "Stopping", "StopFailed", "Deleting", "DeleteFailed"],
+          "x-ms-enum": {
+            "name": "state",
+            "modelAsString": true
+          }
+        },
+        "nodes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnterpriseChannelNode"
+          },
+          "description": "The nodes associated with the Enterprise Channel"
+        }
+      },
+      "description": "The parameters to provide for the Enterprise Channel.",
+      "required": [
+        "nodes"
+      ]
+    },
+    "EnterpriseChannelNode": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of Enterprise Channel Node. This is generated by the Bot Framework.",
+          "readOnly": true
+        },
+        "state": {
```

Comment:
I see that EnterpriseChannelNode and EnterpriseChannelProperties both have a "state" parameter defined by an enum. Is the intention that these enums are the same? Currently, they both have the name "state", so when generated to a client - they are both referencing the same enum model. Does this sound correct to you? (i.e. both models will use exactly the same values for this parameter).

Reply:
That seemed right so I did that. Let me know if I went the wrong way

Reply:
Yes that's correct - changing "nodeState" to something specific should fix it :)

Reply:
Changed it to EnterpriseChannelNodeState